pkg/errors/error.go

```diff
@@ -14,19 +13,12 @@ type Error struct {
 	Details string `json:"details"`
 }
 
-// EncodeError encodes a json error response.
-func EncodeError(ctx *gin.Context, err error, code int, details string) {
-	// construct an error.
-	errorStruct := &Error{
+// AbortWithError stops the chain, writes the status code and the given error
```

Comment:
since this code relies on github.com/gin-gonic/gin, shouldn't we move it in the `pkg/controller` pkg instead? it's not supposed to be used anywhere else, is it?

Reply:
I think we will add more stuff to error package later. Maybe our own error types (for wrapping errors). Error checks, etc. So it will be used outside controllers too. So, I wouldn't move it to controller in this PR and leave it where it is already now.

Reply:
ok. I'm just concerned that this kind of error should only be used as the controller level.